src/app_service/service/wallet_account/service_account.nim

```diff
@@ -306,15 +306,20 @@ proc updateAccountInLocalStoreAndNotify(self: Service, address, name, colorId, e
     if notify:
       self.events.emit(SIGNAL_WALLET_ACCOUNT_POSITION_UPDATED, Args())
 
-proc updatePreferredSharingChainsAndNotify(self: Service, address, prodPreferredChains, testPreferredChains: string) =
+proc updateTestPreferredSharingChainsAndNotify(self: Service, address, testPreferredChains: string) =
```

Comment:
Instead of having `updateTestPreferredSharingChainsAndNotify` and `updateProdPreferredSharingChainsAndNotify` we could have a single one, optimized function like
`proc updatePreferredSharingChainsAndNotify(self: Service, address, preferredChains: string) =` and in the body of this function you can check `if self.areTestNetworksEnabled()` to decide whether to store to `account.testPreferredChainIds` or `account. prodPreferredChainIds`.